client/my-sites/controller.js

```diff
@@ -416,13 +416,14 @@ export function noSite( context, next ) {
 	const hasSite = currentUser && currentUser.visible_site_count >= 1;
 	const isDomainOnlyFlow = context.query?.isDomainOnly === '1';
 	const isJetpackCheckoutFlow = context.pathname.includes( '/checkout/jetpack' );
+	const isGiftCheckoutFlow = context.pathname.includes( '/gift/' );
 
-	if ( ! isDomainOnlyFlow && ! isJetpackCheckoutFlow && hasSite ) {
+	if ( ! isDomainOnlyFlow && ! isJetpackCheckoutFlow && ! isGiftCheckoutFlow && hasSite ) {
 		siteSelection( context, next );
-	} else {
-		context.store.dispatch( setSelectedSiteId( null ) );
-		return next();
 	}
+
+	context.store.dispatch( setSelectedSiteId( null ) );
+	return next();
```

Comment:
What's the significance of moving this out of the else block?

Reply:
No big significance. The `else` just seemed unnecessary to me and I [like to remove them when possible](https://payton.codes/2019/01/22/alternatives-to-else/).

Reply:
The code in the else block now executes in both true and false cases. I just wanted to make sure this was intended and, if it was intended, why that code doesn't have any unintended side-effects.

Reply:
Oh! That was definitely a mistake. Thanks for noticing that.